go-controller/pkg/clustermanager/dnsnameresolver_test.go

```diff
@@ -0,0 +1,483 @@
+package clustermanager
+
+import (
+	"context"
+	"time"
+
+	"github.com/miekg/dns"
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	ocpnetworkapiv1alpha1 "github.com/openshift/api/network/v1alpha1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	egressfirewallapi "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressfirewall/v1"
+	"github.com/urfave/cli/v2"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var _ = ginkgo.Describe("Cluster manager DNS Name Resolver operations", func() {
+	var (
+		app    *cli.App
+		fakeCM *FakeClusterManager
+	)
+
+	ginkgo.BeforeEach(func() {
+		// Restore global default values before each testcase.
+		config.PrepareTestConfig()
+		// Enable egress firewall and dns name resolver.
+		config.OVNKubernetesFeature.EnableEgressFirewall = true
+		config.OVNKubernetesFeature.EnableDNSNameResolver = true
+
+		app = cli.NewApp()
+		app.Name = "test"
+		app.Flags = config.Flags
+
+		fakeCM = NewFakeClusterManagerOVN()
```

Comment:
this file probably should be place in the `dnsnameresolver` folder, and you can try to do something similar to https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/clustermanager/status_manager/status_manager_test.go#L145-L149 (you just need to update https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/util/fake_client.go#L30 with the new DNSNameResolver type)

Reply:
Ok. I'll make the changes.